lib/internal/modules/esm/default_resolve.js

```diff
@@ -91,6 +92,10 @@ function resolve(specifier, parentURL) {
       throw new ERR_UNKNOWN_FILE_EXTENSION(url.pathname);
   }
 
+  if (format === 'cjs') {
+    url.search = '';
+  }
```

Comment:
Looks perfect to me.

Reply:
this should also do `url.hash = '';` I think